services/workflows-service/src/business-report/business-report.service.ts

```diff
@@ -95,11 +109,103 @@ export class BusinessReportService {
+      projectIds,
+    );
+  }
+
   async updateById(...args: Parameters<BusinessReportRepository['updateById']>) {
     return await this.businessReportRepository.updateById(...args);
   }
 
   async count(args: Parameters<BusinessReportRepository['count']>[0], projectIds: TProjectIds) {
     return await this.businessReportRepository.count(args, projectIds);
   }
+
+  async processBatchFile({
+    file,
+    type,
+    countryCode,
+    projectId,
+  }: {
+    file: Express.Multer.File;
+    type: BusinessReportType;
+    countryCode?: string | undefined;
+    projectId: TProjectId;
+  }) {
+    const businessReportsRequests = await parseCsv(file, BusinessReportRequestSchema);
+
+    if (businessReportsRequests.length > 100) {
+      throw new UnprocessableEntityException('Batch size is too large');
+    }
+
+    const batchId = randomUUID();
+    await this.prisma.$transaction(async transaction => {
+      const businessCreatePromise = businessReportsRequests.map(async businessReportRequest => {
+        let business =
+          businessReportRequest.correlationId &&
+          (await this.businessService.getByCorrelationId(businessReportRequest.correlationId, [
+            projectId,
+          ]));
+
+        business ||= await this.businessService.create(
+          {
+            data: {
+              ...(businessReportRequest.correlationId
+                ? { correlationId: businessReportRequest.correlationId }
+                : {}),
+              companyName: businessReportRequest.parentCompanyName || '',
+              projectId,
+            },
+          },
+          transaction,
+        );
+
+        const businessReport = await this.businessReportRepository.create({
+          data: {
+            type,
+            status: BusinessReportStatus.new,
+            report: {},
+            businessId: business.id,
+            batchId,
+            projectId,
+          },
+        });
+
+        return {
+          businessReport: businessReport,
+          businessReportRequest: businessReportRequest,
+          businessId: business.id,
+        } as const;
+      });
+
+      const businessWithRequests = await Promise.all(businessCreatePromise);
+
+      const businessReportRequests = businessWithRequests.map(
+        ({ businessReport, businessReportRequest }) => {
+          return {
+            callbackUrl: `${env.APP_API_URL}/api/v1/internal/business-reports/hook?businessId=${businessReport.businessId}&businessReportId=${businessReport.id}`,
+            websiteUrl: businessReportRequest.websiteUrl,
+            countryCode,
+            parentCompanyName: businessReportRequest.parentCompanyName,
+            lineOfBusiness: businessReportRequest.lineOfBusiness,
+            businessReportId: businessReport.id,
+          };
+        },
+      ) satisfies TReportRequest;
+
+      await new UnifiedApiClient().postBatchBusinessReport({
+        reportRequests: businessReportRequests,
+        clientName: 'merchant',
```

Comment:
It should also get manual review flag

Reply:
don't think so. it should be automatic and fast